lessons/4_numbers/index.html

```diff
@@ -0,0 +1,275 @@
+    nameEl.textContent = 'Name: ' + event.target.value;
+});
+                    </code>
+                </pre>
+                <div class="input_group">
+                    <div class="input-group__input-wrapper"><input id="name_input" placeholder="enter your name" /></div>
+                    <p id="name">Name: </p>
+                </div>
+                <style>
+                    .input_group {
+                        display: flex;
+                        width: 100%;
+                        justify-content: center;
+                    }
+                    .input-group__input-wrapper {
+                        display: flex;
+                        justify-content: center;
+                        flex-direction: column;
+                        margin: 0 1em !important;
+                    }
+                </style>
+                <script>
+                    (() => {
+                        const nameInputEl = document.getElementById('name_input');
+                        const nameEl = document.getElementById('name');
+
+                        nameInputEl.addEventListener('change', function(event) {
+                            nameEl.textContent = 'Name: ' + event.target.value;
+                        });
+                    })();
+                </script>
+            </section>
+            <section>
+                <h3>basic arithmetic</h3>
+                <pre>
+                    <code>
+// Some basic arithmetic
+1 + 1; // = 2
+0.1 + 0.2; // = 0.30000000000000004
+8 - 1; // = 7
+10 * 2; // = 20
+35 / 5; // = 7
+
+// uneven division.
+5 / 2; // = 2.5
+
+// And modulo division.
+10 % 2; // = 0
+30 % 4; // = 2
+18.5 % 7; // = 4.5
+
+// Precedence is enforced with parentheses.
+(1 + 3) * 2; // = 8
+                    </code>
+                </pre>
+            </section>
+            <section>
+                <h3>Add calculator</h3>
+                <div class="input_group">
+                    <div class="input-group__input-wrapper"><input id="add_one_input" placeholder="number" /></div>
+                    <p>+</p>
+                    <div class="input-group__input-wrapper"><input id="add_two_input" placeholder="number" /></div>
+                    <p>=</p>
+                    <p id="add_answer"></p>
+                </div>
+                <pre>
+                    <code>
+const inputOneEl = document.getElementById('add_one_input');
+const inputTwoEl = document.getElementById('add_two_input');
+const answerEl = document.getElementById('add_answer');
+
+function handleInputChange() {
+    const answer = inputOneEl.value + inputTwoEl.value;
+    answerEl.textContent = answer;
+}
+inputOneEl.addEventListener('change', handleInputChange);
+inputTwoEl.addEventListener('change', handleInputChange);
+                    </code>
+                </pre>
+                <p class="fragment" style="color: red;">Oh no, it is adding strings together</p>
```

Comment:
Will you explain why it's happening in details and provide more examples later like 2 + '2', '2' + 2 and so on?

Reply:
Added slide to explain why.